In [1]:
#Run this
import pandas as pd                    # imports pandas and calls the imported version 'pd'
import matplotlib.pyplot as plt        # imports the package and calls it 'plt'
import seaborn as sns                  # imports the seaborn package with the imported name 'sns'
sns.set()  

## Steam Data

Steam is the world's most popular PC Gaming hub. They have a massive catalog of games, with everything from AAA blockbusters to small indie titles.  


Note that for each user-game combo, there will either be one row (if the game has only been purchased but not played) or two rows (if the game has been both purchased and played).  For example, if I bought Portal, a row would be added to the data representing the purchase of that game by me.  When I open the game and start playing it, a second row would be added to the data representing that I have played the game along with recording my play time for that specific game.


In [3]:
df = pd.read_csv('steam_sample.csv')

In [35]:
df.describe()

user_id
count  7.806000e+03
mean   1.062334e+08
std    6.764807e+07
min    6.830190e+05
25%    4.879807e+07
50%    9.907790e+07
75%    1.574670e+08
max    3.092654e+08

In [25]:
df.head(5)

user_id  game_name  activity hours_played_if_play
0  308653033   Unturned  purchase                    1
1  308653033   Unturned      play                  0.6
2  308653033  theHunter  purchase                    1
3  144004384     Dota 2  purchase                    1
4  144004384     Dota 2      play                   22

In [28]:
df.dtypes

user_id                  int64
game_name               object
activity                object
hours_played_if_play    object
dtype: object

### 2. Preparing the Data

In order to prepare the data effectively, we need to clean the data:

1. Identify any values that have been encoded in the csv to represent a **missing value**.    
4. Drop any observations with missing values.

In [29]:
df['hours_played_if_play'].describe()

count     7806
unique     473
top          1
freq      4976
Name: hours_played_if_play, dtype: object

In [31]:
missing_values = df['hours_played_if_play'].unique()

array(['1', '0.6', '22', '1028', '1008', '148', '108', '72', '36', '35',
       '32', '21', '16', '15.8', '8.6', '7.8', '7.3', '3.1', '1.9', '1.7',
       '1.1', '0.4', '153', '63', '26', '1.4', '639', '479', '70', '65',
       '33', '30', '19.8', '16.2', '11.3', '4.2', '3.9', '2.3', '0.8',
       '0.7', '0.5', '0.3', '396', '227', '13.4', '12.6', '11.2', '10.1',
       '2.4', '210', '1.2', '0.2', '13.2', '48', '110', '0.1', '67',
       '429', '5.5', '61', '1.6', '18.3', '9.9', '4.7', '1714', '441',
       '197', '147', '117', '86', '73', '49', '46', '31', '24', '20',
       '19.7', '18.2', '14.2', '11.6', '9.8', '9.7', '8.4', '6.5', '4.8',
       '4.1', '3.7', '2.9', '2.7', '2.1', '222', 'unknown', '14.9',
       '14.1', '83', '11.1', '3.2', '6.9', '395', '251', '9.3', '7.4',
       '54', '34', '1.8', '99', '98', '96', '29', '27', '23', '19.1',
       '18.7', '17.5', '17', '16.6', '14.7', '13.3', '10.7', '10.2', '10',
       '9.6', '9.5', '8', '7.9', '7.6', '7', '6.6', '6', '5.8', '5

In [34]:
dfdrop = pd.read_csv('steam_sample.csv', na_values = 'unknown')
dfdrop

user_id                                    game_name  activity  \
0     308653033                                     Unturned  purchase   
1     308653033                                     Unturned      play   
2     308653033                                    theHunter  purchase   
3     144004384                                       Dota 2  purchase   
4     144004384                                       Dota 2      play   
...         ...                                          ...       ...   
7801   99096740                             SimCity 4 Deluxe      play   
7802   99096740  BioShock Infinite Burial at Sea - Episode 2  purchase   
7803   99096740       The Elder Scrolls V Skyrim - Dawnguard  purchase   
7804   99096740      The Elder Scrolls V Skyrim - Dragonborn  purchase   
7805   99096740      The Elder Scrolls V Skyrim - Hearthfire  purchase   

      hours_played_if_play  
0                      1.0  
1                      0.6  
2                      1.0  
3                      1.0  
4                     22.0  
...                    ...  
7801                   0.2  
7802                   1.0  
7803                   1.0  
7804                   1.0  
7805                   1.0  

[7806 rows x 4 columns]

In [37]:
dfdrop = dfdrop.dropna()
dfdrop

user_id                                    game_name  activity  \
0     308653033                                     Unturned  purchase   
1     308653033                                     Unturned      play   
2     308653033                                    theHunter  purchase   
3     144004384                                       Dota 2  purchase   
4     144004384                                       Dota 2      play   
...         ...                                          ...       ...   
7801   99096740                             SimCity 4 Deluxe      play   
7802   99096740  BioShock Infinite Burial at Sea - Episode 2  purchase   
7803   99096740       The Elder Scrolls V Skyrim - Dawnguard  purchase   
7804   99096740      The Elder Scrolls V Skyrim - Dragonborn  purchase   
7805   99096740      The Elder Scrolls V Skyrim - Hearthfire  purchase   

      hours_played_if_play  
0                      1.0  
1                      0.6  
2                      1.0  
3                      1.0  
4                     22.0  
...                    ...  
7801                   0.2  
7802                   1.0  
7803                   1.0  
7804                   1.0  
7805                   1.0  

[7800 rows x 4 columns]

In the original df there are 7806 rows, through dropping 'unknown' we get 7800 rows so 6 rows were dropped. 

### 3. Interpreting Missing Data

We dropped observations above that had any missing values. Unless the creator of the data set did not set another value for missing data other than "unknown" we did the best way to handle missing data

### 4. Separate the Data

In this analysis, we would like to answer questions based on purchases and based on play time.  Create two new dataframes:

- one that is comprised of the purchase rows for the games that were purchased
- one that is comprised of the play rows for the games that were played

In [41]:
purchased = dfdrop[dfdrop['activity'] == 'purchase']
purchased

user_id                                    game_name  activity  \
0     308653033                                     Unturned  purchase   
2     308653033                                    theHunter  purchase   
3     144004384                                       Dota 2  purchase   
5      54103616              Counter-Strike Global Offensive  purchase   
7      54103616                               Counter-Strike  purchase   
...         ...                                          ...       ...   
7800   99096740                             SimCity 4 Deluxe  purchase   
7802   99096740  BioShock Infinite Burial at Sea - Episode 2  purchase   
7803   99096740       The Elder Scrolls V Skyrim - Dawnguard  purchase   
7804   99096740      The Elder Scrolls V Skyrim - Dragonborn  purchase   
7805   99096740      The Elder Scrolls V Skyrim - Hearthfire  purchase   

      hours_played_if_play  
0                      1.0  
2                      1.0  
3                      1.0  
5                      1.0  
7                      1.0  
...                    ...  
7800                   1.0  
7802                   1.0  
7803                   1.0  
7804                   1.0  
7805                   1.0  

[4928 rows x 4 columns]

In [43]:
played = dfdrop[dfdrop['activity'] == 'play']
played

user_id                        game_name activity  \
1     308653033                         Unturned     play   
4     144004384                           Dota 2     play   
6      54103616  Counter-Strike Global Offensive     play   
8      54103616                   Counter-Strike     play   
10     54103616                      Left 4 Dead     play   
...         ...                              ...      ...   
7793   99096740                           Crysis     play   
7795   99096740              Assassin's Creed II     play   
7797   99096740               Hitman Blood Money     play   
7799   99096740     The Binding of Isaac Rebirth     play   
7801   99096740                 SimCity 4 Deluxe     play   

      hours_played_if_play  
1                      0.6  
4                     22.0  
6                   1028.0  
8                   1008.0  
10                   148.0  
...                    ...  
7793                   5.3  
7795                   2.7  
7797                   1.3  
7799                   0.7  
7801                   0.2  

[2872 rows x 4 columns]

### 5. Game Play Time Question

Amount of time played for each row. 



In [51]:
played['hours_played_if_play'].sort_values(ascending=False)

5106    11754.0
531      4814.0
1592     3626.0
769      3309.0
4278     3178.0
         ...   
7175        0.1
7157        0.1
4179        0.1
4177        0.1
3125        0.1
Name: hours_played_if_play, Length: 2872, dtype: float64

Over what period of time was this data collected? If this is data from the past year someone spent 489 days playing a game which is not possible
Over the span of 2 years someone spent over half their 2 years playing a game which is worrisome


### 6.Purchasing to Playing?

Overall, of all of the purchases represented in this data, what proportion have been played?  Of all the purchases in the data, what proportion remain "unopened", that is, unplayed?

What factors can we anticipate might affect whether a given game is played or not after being purchased?

In [54]:
len(played)/len(purchased)

0.5827922077922078

58.3 percent have been played while 41.7 percent remain unoppened. A user has other games that they are playing or just forgot that they purchased. Or did not mean to purchase. 